haskell-language-server.cabal

```diff
@@ -111,49 +109,55 @@ flag cabalfmt
 common cabalfmt
   if flag(cabalfmt)
     build-depends: haskell-language-server:hls-cabal-fmt-plugin
-    cpp-options: -Dhls_cabalfmt
+    cpp-options:   -Dhls_cabalfmt
 
 flag isolateCabalfmtTests
-  description: Should tests search for 'cabal-fmt' on the $PATH or shall we install it via build-tool-depends?
+  description:
+    Should tests search for 'cabal-fmt' on the $PATH or shall we install it via build-tool-depends?
+
   -- By default, search on the PATH
   default:     False
   manual:      True
 
 library hls-cabal-fmt-plugin
-  import:           defaults, pedantic, warnings
+  import:          defaults, pedantic, warnings
+
   if !flag(cabalfmt)
     buildable: False
-  exposed-modules:  Ide.Plugin.CabalFmt
-  hs-source-dirs:   plugins/hls-cabal-fmt-plugin/src
+
+  exposed-modules: Ide.Plugin.CabalFmt
+  hs-source-dirs:  plugins/hls-cabal-fmt-plugin/src
   build-depends:
-    , base            >=4.12 && <5
+    , base            >=4.12    && <5
     , directory
     , filepath
-    , ghcide          == 2.8.0.0
-    , hls-plugin-api  == 2.8.0.0
+    , ghcide          ==2.8.0.0
+    , hls-plugin-api  ==2.8.0.0
     , lens
     , lsp-types
     , mtl
     , process-extras
     , text
 
 test-suite hls-cabal-fmt-plugin-tests
-  import:           defaults, pedantic, test-defaults, warnings
+  import:         defaults, pedantic, test-defaults, warnings
+
   if !flag(cabalfmt)
     buildable: False
-  type:             exitcode-stdio-1.0
-  hs-source-dirs:   plugins/hls-cabal-fmt-plugin/test
-  main-is:          Main.hs
+
+  type:           exitcode-stdio-1.0
+  hs-source-dirs: plugins/hls-cabal-fmt-plugin/test
+  main-is:        Main.hs
   build-depends:
     , base
     , directory
     , filepath
     , haskell-language-server:hls-cabal-fmt-plugin
-    , hls-test-utils        == 2.8.0.0
+    , hls-test-utils                                ==2.8.0.0
 
-  if flag(isolateCabalfmtTests)
+  if flag(isolatecabalfmttests)
```

Comment:
TIL Cabal converts flag names to lowercase.

Reply:
This is weird, is that `cabal-fmt` preference? It seems like `cabal-gild` doesn't convert to lower case.

Reply:
Gild currently doesn't convert flags to lowercase, but it probably will after I fix https://github.com/tfausak/cabal-gild/issues/72. Conditionals in Cabal are represented by [`Condition`](https://hackage.haskell.org/package/Cabal-syntax-3.12.0.0/docs/Distribution-Types-Condition.html#t:Condition) [`ConfVar`](https://hackage.haskell.org/package/Cabal-syntax-3.12.0.0/docs/Distribution-Types-ConfVar.html#t:ConfVar), which can have a [`FlagName`](https://hackage.haskell.org/package/Cabal-syntax-3.12.0.0/docs/Distribution-Types-Flag.html#t:FlagName) in them. I can't currently find any documentation about it, but flag names are explicitly converted to lowercase when parsing: https://github.com/haskell/cabal/blob/03d98294e1c071be368f04dbe6fc9cb263bea86d/Cabal-syntax/src/Distribution/Types/Flag.hs#L117

Reply:
There's some discussion here (and in linked issues): https://github.com/haskell/cabal/issues/5043

Reply:
FWIW I think that just because cabal compares flag names case-insensitively, that's not necessarily a reason to normalize the user's flag name to lower-case. And it would be fairly unwelcome...

Reply:
Starting with [version 1.3.1.3](https://github.com/tfausak/cabal-gild/releases/tag/1.3.1.3), Gild now behaves the same as `cabal-fmt` in that it converts flag names into lowercase. This is a consequence of going through the [`FlagName` type](https://hackage.haskell.org/package/Cabal-syntax-3.12.1.0/docs/Distribution-Types-Flag.html#t:FlagName) from `Cabal-syntax`.